### ISSUE TITLE
Sticky Blocks

### ISSUE THREAD

**Issue by C1ffisme** [minetest-mods/mesecons#218]
This is a feature request, not a bug.

I really wish we had some sticky blocks, like in minecraft. It would allow us to build those cool flying machines people build, or we could more asthetic piston gates.

We could make slime/glue blocks, but I would rather prefer something like cobble with glue on it. That would allow people to make good-looking piston gates, with a double-sticky piston.

**Comment by Jeija** [comment]
Sounds like a good idea. You could achieve some of this with a sticky movestone already (which pulls the whole stack of blocks), but I see how this could work much better. I didn't even know something like this existed in minecraft.
Instead of just copying the minecraft idea: Do you / anyone else have an idea on how to allow for flying machines etc. in another, mesecon-like way? If we go with sticky blocks, I also prefer the cobble idea btw.

**Comment by HybridDog** [comment]
>It would allow us to build those cool flying machines people build

You can already make flying machines with technic frames.
l put lava and water on it and let it move and then l had a sky path machine… until the admin of the server banned me…

**Comment by C1ffisme** [comment]
@HybridDog The piston idea sounds cool, but about the flying machine, how did it get you banned?

**Comment by HybridDog** [comment]
The lava cooled on the ground, too. And so a greenhouse of someone became covered.

**Comment by C1ffisme** [comment]
Bump.

**Comment by Jeija** [comment]
I just made a very basic implementation of sticky blocks and pushed it to https://github.com/Jeija/minetest-mod-mesecons/tree/sticky_blocks. It does not have any textures yet, and right now there is only one kind of sticky block. I'd like to have multiple sticky blocks, e.g. just a block that is sticky on two faces (to make corners) or a T-junction sticky block.
Sticky blocks are a mess, and bugs connected with sticky blocks and pistons are hard to fix. Therefore, if sticky blocks will ever end up in mesecons, there are a lot of bugfixes to do. If you feel like you could spot and fix the bugs in that branch, it would be great if you could help. Because I won't be able to spend that much time on fixing piston-related bugs for cornercases.

**Comment by HybridDog** [comment]
Could you add a mesecons.register_on_piston_push(pos1, pos2, …)?

**Comment by C1ffisme** [comment]
@Jeija I tried the build. Currently the block just acts like a normal block, but that might be because I put the sticky block mod into my mesecons folder and chucked out the rest of the newly downloaded stuff. Lemme try with the new stuff merged with my mesecons (I keep some of my own mods in there.)

**Comment by C1ffisme** [comment]
![screenshot_3503245222](https://cloud.githubusercontent.com/assets/9893266/6426659/8e5fc5d4-bf2c-11e4-8f4f-681eca9d3e5a.png)

This happened. LOL.

**Comment by Jeija** [comment]
Yes, it is because of these kinds of bugs that I can't merge the sticky blocks into the minetest-mod-mesecons master. It would be great if you could help me fix them, they are very tricky ones.

**Comment by C1ffisme** [comment]
I found another one, which might or might not be easier to fix:
![screenshot_3594586733](https://cloud.githubusercontent.com/assets/9893266/6431782/76b0b2c4-c001-11e4-9cec-fd84f947b286.png)

And then:

![screenshot_3594601606](https://cloud.githubusercontent.com/assets/9893266/6431783/82479026-c001-11e4-9553-4dbad6a618cd.png)

Blocks attached to the piston do not push entities.

**Comment by benrob0329** [comment]
In the current implementation, are any blocks not effected by the all sides sticky block? (For eg Obsidian?)

**Comment by Jeija** [comment]
IIRC there are only the MVPS stoppers (like extended pistons) that completely stop the push / pull action, I don't think single blocks can not be effected in the current implementation.

**Comment by JamesTobin** [comment]
https://youtu.be/xFJEDV8WIns

**Comment by HybridDog** [comment]
JamesTobin, you can make such a machine without sticky blocks, too.

**Comment by JamesTobin** [comment]
@HybridDog  show me how please

**Comment by HybridDog** [comment]
Unfortunately l don't know (or didn't understand) how to do it with pure mesecons, but some time ago rubenwardy (l think it was rubenwardy) told me he/she did it. l think l was in a nether biome when rubenwardy told me that you can make a digging mesecons vehicle without technic frames and just pure mesecons, he/she wrote something about using movestones for the machine. l have hardly any experience with movestones.

**Comment by JamesTobin** [comment]
I've used them b4. anyways, these sticky blocks are definitely nice!

**Comment by Wuzzy2** [comment]
Another bug I found: If you build a chain of sticky blocks which are placed in a way that would push the piston itself, the final sticky block disappears.

Before:
https://i.imgur.com/E29MeFO.png
After:
https://i.imgur.com/TWFHb4A.png

Note the different number of stcky blocks. One block got destroyed.

I suggest to simply prevent pushing if such a build appears. This would be the simplest solution.

For your interest: The Minecraft equivalent of sticky blocks are slime blocks:
https://minecraft.gamepedia.com/Slime_Block


Suggested crafting recipe (when this is finished):

```
GGG
GWG
GGG
```
* G = glue
* W = any wood

**Comment by atoms118** [comment]
Bump?

**Comment by numberZero** [comment]
Sticky blocks are there since ages. There is still no crafting recipe but in creative mode they’re usable. Let’s discuss individual bugs separately.
